util_test.go

```diff
@@ -0,0 +1,158 @@
+package pokemonbattlelib
+
+import (
+	"errors"
+	"fmt"
+	"reflect"
+	"strings"
+
+	"github.com/onsi/gomega/types"
+)
+
+// Gomega custom matchers, see: https://onsi.github.io/gomega/#adding-your-own-matchers
+
+// Used for custom gomega matchers. Checks to see if a is probably the same pokemon as b based on values that are unlikely to change as a result of a transaction.
+func comparePokemon(a, b *Pokemon) bool {
+	return a.NatDex == b.NatDex &&
+		a.Nature.name == b.Nature.name &&
+		a.Gender == b.Gender &&
+		a.Elemental == b.Elemental
+}
```

Comment:
I guess we can't just compare a pointer here because copies are made? Is it worth having a global Pokemon ID/hash attached to the Pokemon to make this comparison safer?

Reply:
It would probably be easier (and faster) to just always refer to a pokemon by party/slot. Or we could change the `target` type so that we can have a pointer to the original pokemon, changing it after getting the `Turn` from the agent.

Reply:
I think the first point makes sense, but I think the user should still be able to see the Pokemon without having to get it separately when making a turn. I don't think we should change pointers because that could lead to unexpected side effects, especially if something related to turns changes. Can't we just assume that if the party/slot match, the target is correct in the tests?

Reply:
> unexpected side effects, especially if something related to turns changes

Like what? If we just change the pointer back to the original copy, how would that affect other behavior?

> Can't we just assume that if the party/slot match, the target is correct in the tests?

Maybe? That feels kinda brittle to me, especially considering that we had #127.

Reply:
I think I just need to see an implementation before judging this. Go with whatever your intuition is and it will probably be fine.

Reply:
Can we go ahead and merge this then?